File: DependencyInjection/ConformingContainerAntipattern/_3_ConformingContainer/Core/TeleComSystem.cs
using ConformingContainerAntipattern._3_ConformingContainer.Inbound;
using ConformingContainerAntipattern._3_ConformingContainer.Outbound;

namespace ConformingContainerAntipattern._3_ConformingContainer.Core
{
  class TeleComSystem
  {
    private readonly IProcessingWorkflow _processingWorkflow;
    private readonly IInbound _inbound;
    private readonly IOutbound _outbound;

    public TeleComSystem()
    {
      _inbound = ApplicationRoot.Context.Resolve<IInbound>();
      _outbound = ApplicationRoot.Context.Resolve<IOutbound>();
      _processingWorkflow = ApplicationRoot.Context.Resolve<IProcessingWorkflow>();
    }

    public void Start()
    {
      _inbound.SetDomainLogic(_processingWorkflow);
      _processingWorkflow.SetOutbound(_outbound);
      _inbound.StartListening();
    }
  }
}

File: Java/dependency-injection/conforming-container-antipattern/src/main/java/com/github/grzesiek_galezowski/ConformingContainerAntipattern/Core/TeleComSystem.java
package com.github.grzesiek_galezowski.ConformingContainerAntipattern.Core;

import com.github.grzesiek_galezowski.ConformingContainerAntipattern.ApplicationRoot;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Inbound.Inbound;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Outbound.Outbound;

public class TeleComSystem {
  private final ProcessingWorkflow _processingWorkflow;
  private final Inbound _inbound;
  private final Outbound _outbound;

  public TeleComSystem() {
    _inbound = ApplicationRoot.CONTEXT.resolve(Inbound.class);
    _outbound = ApplicationRoot.CONTEXT.resolve(Outbound.class);
    _processingWorkflow = ApplicationRoot.CONTEXT.resolve(ProcessingWorkflow.class);
  }

  public void start() {
    _inbound.setDomainLogic(_processingWorkflow);
    _processingWorkflow.setOutbound(_outbound);
    _inbound.startListening();
  }
}
